commands/command_pre_push.go

```diff
@@ -137,7 +133,19 @@ func prePushRef(left, right string) {
 			os.Exit(2)
 		}
 	}
+}
+
+func filteredPointers(pointers []*lfs.WrappedPointer, filter lfs.StringSet) []*lfs.WrappedPointer {
```

Comment:
How about this? https://gist.github.com/technoweenie/34671a1296ba375b477a

```bash
$ go test -bench . -run BenchmarkFilterByChan
testing: warning: no tests to run
PASS
BenchmarkFilterByList-4    	    1000	   1871779 ns/op
BenchmarkFilterByCallback-4	    1000	   1653444 ns/op
BenchmarkFilterByChan-4    	     200	   9165917 ns/op
```

Still leaning towards the current approach. It's not much slower than using a callback, and has the simplest implementation. Unless there's some radical way to make the channel approach simpler and faster, I'm not even really considering it.